Initial Work.

bscutt opened the pull request:
Initial work from the tentacles editor, tidied with comments and some bugs fixed.

bscutt committed:
Initial Work.

Initial work from the tentacles editor, tidied with comments and some bugs fixed.

bscutt committed:
Remove browser prefixing, as this is added in the build process.

Remove mix-manifest file, as this is unused.

bscutt committed:
Rename plugin.

jonathanbull commented on src/scss/emoji-picker.scss:
```diff
@@ -0,0 +1,59 @@
+$highlight: #81409b;
+
+.emoji-picker {
```
Should we match the 4 space indentation used in the other files?

bscutt committed:
Reduce the set of emoji used - don't add any not supported by iOS11

bscutt committed:
Use 4 spaces for indentation.

bscutt committed:
Rename the plugin to 'emoticons'.

Rename the plugin to 'emoticons', so it can be used as a drop in replacement for the existing plugin. Add a title element to the emoji spans.

bscutt commented on src/js/emoji-picker.js:
```diff
@@ -0,0 +1,237 @@
+/*!
+ * EmailOctopus EmojiPicker v1.0.0 (https://emailoctopus.com/)
+ * License MIT
```
@jonathanbull - we should decide what license we're going to use (if we're going to open source it) - MIT is the most common one.

jonathanbull replied:
MIT is good for me

jonathanbull commented:
If we are making this a direct replacement for Froala's emoticons library, I think we should also name our `css` and `js` files to match:

![screen shot 2018-07-24 at 10 58 29](https://user-images.githubusercontent.com/1151115/43131264-86c66008-8f30-11e8-91bd-180f4a104ef4.png)

so that they simply need to switch `plugins` to `third_party` in the file path. Otherwise we can't really advertise it as a drop-in replacement.

But then that _would_ mean we have to name the files `emoticons.*`, which is a bit lame as that's not the name of our project. I'm beginning to think we should just set the ID to `emojis` to keep things simple and differentiate our plugin - the process of adding plugins is quite simple so I don't think people will struggle. It also avoids any expectations that the Froala `emoticon` settings will apply to our plugin (which even if they currently do, may not work in the future).

What do you think? Either way we should add a quick `README.md` (sorry, know it's boring).

bscutt committed:
Change name.

bscutt committed:
Add readme and license details.
